### PR TITLE
feat: consolidate post-reference file bump actions into one workflow + populate automated replaced by map in jinja template

### PR DESCRIPTION
**Pull request by nayib-jose-gloria**
Changes: 
- Populate replaced_by_map in ontology dry run and write to JSON, refactor script to produce 2 output files
- Update generate migrate script job to retrieve replaced_by_map JSON produced by ontology dry run, if applicable. 
- Update ontology dry run job to store replaced_by_map in github cache, for access in generate migrate script job
      -  github caches auto-delete after 7 days without access, cache will be clear by the time of the next monthly run (or can be deleted manually if needed) 
- Consolidate all the post-reference file update jobs (publish to pypi, gencode dry run, ontology dry run, generate migrate script) into one 'minor schema update' workflow. 
- Introduce steps to check which files changed, to determine whether gencode, ontology, or both dry runs need to run
- generate migrate script job is dependant on ontology dry run (if applicable) and publish to pypi jobs
- ontology dry run, gencode dry run, and publish to pypi jobs run in parallel

### CODE REVIEW

File: .github/workflows/minor-schema-version-bump.yml [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+#      - name: Publish distribution to PyPI
+#        if: startsWith(github.ref, 'refs/tags')
+#        uses: pypa/gh-action-pypi-publish@release/v1
+#        with:
+#          password: ${{ secrets.PYPI_API_TOKEN }}
+#          package-dir: cellxgene_schema_cli/
+
+  generate-conversion-script:
+    runs-on: ubuntu-latest
+    needs: [ontology-dry-run, publish-to-pypi]
```
Comment (Bento007) [inline .github/workflows/minor-schema-version-bump.yml:160]: this should only depend on publish-to-pypi

**Review by Bento007 — CHANGES_REQUESTED** [review]
Looks good, on thing to address.

File: .github/workflows/minor-schema-version-bump.yml (hunk 1 above)
Comment (nayib-jose-gloria) [reply .github/workflows/minor-schema-version-bump.yml:160]: it depends on ontology dry run to populate + write the map of deprecated terms with known replacement terms

**Comment by Bento007** [comment]
these changers will conflict with my exist PR to move files around. landing https://github.com/chanzuckerberg/single-cell-curation/pull/476 first

File: scripts/migration_assistant/tests/test_convert_template.py [hunk 2]
```diff
@@ -98,9 +97,99 @@ def migrate(input_file, output_file, collection_id, dataset_id):
...
     dataset.write(output_file, compression="gzip")"""
-    mock.patch("migration_assistant.generate_script.get_current_version", return_value=expected_output)
 
     with open(os.path.join(mock_target_file), "r") as fp:
         actual_output = fp.read().strip()
 
     assert actual_output == expected_output[1:]
+
+
+def test_generate_script_with_automated_replaced_by_map(template, tmpdir):
```
Comment (Bento007) [inline scripts/migration_assistant/tests/test_convert_template.py:107]:
```suggestion
def test_generate_script__with_automated_replaced_by_map(template, tmpdir):
```

File: scripts/schema_bump_dry_run_ontologies/tests/test_ontology_bump_dry_run.py [hunk 3]
```diff
@@ -94,38 +105,54 @@ def mock_fetch_private_dataset(self, *args):
 
     def test_no_deprecated_terms_no_revisions(self):
         self.private_collections.pop()
-        with NamedTemporaryFile() as tmp, open("fixtures/no_deprecated_terms_no_revisions_expected", "rb") as expected:
-            ontology_bump_dry_run.dry_run(tmp.name)
+        with NamedTemporaryFile() as tmp, NamedTemporaryFile() as tmp_json, open(
```
Comment (Bento007) [inline scripts/schema_bump_dry_run_ontologies/tests/test_ontology_bump_dry_run.py:108]: even though these are unittest, I think you can still use pytest fixtures.

**Review by Bento007 — APPROVED** [review]

**Pull request merged**